indico/util/reporter.py

```diff
@@ -0,0 +1,72 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from copy import deepcopy
+
+from flask import session, request
+
+
+class ReporterBase(object):
+    """Serves as a base class for classes responsible to perform actions on
+    reports
+    """
+
+    #: A unique ID for the report
+    REPORT_ID = ''
+    #: Dict containing the default visible columns and filtering options
+    DEFAULT_REPORT_CONFIG = {}
+    #: OrderedDict containing the filterable columns of the report
+    FILTERABLE_ITEMS = None
+    cache = None
+
+    @classmethod
+    def get_report_config_session_key(cls, entry_parent_id):
+        """Composes the unique configuration ID that will be used as a key to
+        set the report's configuration to the session"""
+        return '{}_config_{}'.format(cls.REPORT_ID, entry_parent_id)
+
+    @classmethod
+    def get_report_config(cls, entry_parent_id):
+        """Get the report's configuration from the session and return it"""
+        session_key = cls.get_report_config_session_key(entry_parent_id)
+        report_config_uuid = request.args.get('config')
+        if report_config_uuid:
+            configuration = cls.cache.get(report_config_uuid)
+            if configuration and configuration['entry_parent_id'] == entry_parent_id:
+                session[session_key] = configuration['data']
+        return session.get(session_key, cls.DEFAULT_REPORT_CONFIG)
+
+    @classmethod
+    def query_report_entries(cls):
```

Comment:
From what I understand in the docstring this method does not query but rather *gets* a query. Why not simply `get_query` instead?